src/main/java/sg/ncl/MainController.java

```diff
@@ -121,8 +123,23 @@ public String resources() {
         return "resources";
     }
 
+    @RequestMapping("/research")
+    public String research() {
+        return "research";
+    }
+
+    @RequestMapping("/calendar")
+    public String calendar() {
+        return "calendar";
+    }
+
+    @RequestMapping("/calendar1")
+    public String calendar1() {
+        return "calendar1";
+    }
+
     @RequestMapping(value="/futureplan/download", method=RequestMethod.GET)
-    public void futureplanDownload(HttpServletResponse response) {
+    public void futureplanDownload(HttpServletResponse response) throws FuturePlanDownloadException {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Document this public method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUndocumentedApi)